MoltenVK/MoltenVK/GPUObjects/MVKVulkanAPIObject.mm

```diff
@@ -27,12 +27,31 @@
 VkResult MVKVulkanAPIObject::setDebugName(const char* pObjectName) {
 	if (pObjectName) {
 		[_debugName release];
-		_debugName = [[NSString alloc] initWithUTF8String: pObjectName];	// retained
+		@autoreleasepool {
+			_debugName = [[NSString alloc] initWithUTF8String: pObjectName];	// retained
+		}
 		propagateDebugName();
 	}
 	return VK_SUCCESS;
 }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wobjc-method-access"
+void MVKVulkanAPIObject::setMetalObjectLabel(id mtlObj, NSString* label) {
+	if ( !label ) { return; }
+
+	// If debug mode is enabled, append the object address.
+	// Since doing so might internally create other Metal objects, so catch them in an autorelease pool.
```

Comment:
```suggestion
	// Doing so might internally create other Metal objects, so catch them in an autorelease pool.
```

Using both "since" and "so" in the same sentence like that feels awkward.

Reply:
Thanks. Fixed.